Danki2/Assets/Scripts/Abilities/InstantCast/Fireball.cs

```diff
@@ -3,16 +3,27 @@
 public class Fireball : InstantCast
 {
     private static readonly float FIREBALL_SPEED = 5;
+    private readonly AbilityContext _context;
 
     public Fireball(AbilityContext context) : base(context)
     {
+        _context = context;
     }
 
     public override void Cast()
     {
         Vector3 position = Context.Owner.transform.position;
         Vector3 target = Context.TargetPosition;
         Quaternion rotation = Quaternion.LookRotation(target - position);
-        FireballObject.Fire(position, rotation, FIREBALL_SPEED);
+        FireballObject.Fire(_context.Owner, OnCollision, FIREBALL_SPEED, position, rotation);
+    }
+
+    public void OnCollision(GameObject gameObject)
+    {
+        if (gameObject.TryGetComponent<Mortal>(out var mortal))
```

Comment:
Just had a little look round to see if we can avoid having to try and get the component every time. I think we should be using tags, see here: https://answers.unity.com/questions/29521/how-to-make-collision-check-if-other-collider-has.html We probably want to give all enemies an "enemy" tag. I'll raise a story for us to think about this next sprint as I don't think we need to do it as part of this PR

Reply:
Raised in Sprint 5

Reply:
Yeah good point - as it stands it's going to be friendly fire alley.